### PR TITLE
Encapsulate a tools for stream load when import file with a big size #4031

### LINKED ISSUE #4031: A import tool to loading large files in the local [General]
### Requirement Description
1. Encapsulate a tool to ingest a CSV file into StarRocks.
2. The requirement is not split the file really but instead read the file by part.

### Mento
@chaoyli

### PR DESCRIPTION
**Pull request by itweixiang**
## What type of PR is this：
- [ ] bug
- [x] feature
- [ ] enhancement
- [ ] others

## Which issues of this PR fixes ：

Fixes #4031

## Problem Summary(Required) ：

Use java and shell realize tools about stream load

The method of stream load is limited by `curl` performance. As a result, it's too slow to import a big file.
I want to realize a Java client to transfer the big file in parallel, can increase the performance.

First, I need to sequential read the big file, due to the characteristics of the disk, the performance of sequential read is of less concern about thread number.
Second, the data that was read will compose a long string, and be put in the blocking queue. when the main thread will block, the size of the blocking queue is equal to a limited size.
Third, the worker thread will consume the blocking queue, and send data through the network. when the worker thread is stopped, they receive an empty string from the main thread.

### For curl

图片

The method, using curl to send data,  is too low in transmission performance. The transmission speed is only tens of megabytes.

### For these tool

图片

With the new tools, the transmission speed can run over the 10 Gigabit network. 

However, the transmission speed may also be affected by the sequential read-write performance of the hard disk. If you want faster transmission speed, you need a hard disk with better performance

### CODE REVIEW

File: tools/stream_load/StreamLoadImportUtils.java [hunk 1]
```diff
@@ -0,0 +1,220 @@
...
+            inputStream.close();
+            service.shutdown();
+        } catch (IOException e) {
+            e.printStackTrace();
+        }
+    }
+
+    private static void printConfig() {
+        if (url == null) {
+            System.out.println("url not empty , please set --url=xxx");
```
Comment (chaoyli) [inline tools/stream_load/StreamLoadImportUtils.java:68]: url is empty

File: tools/stream_load/StreamLoadImportUtils.java [hunk 2]
```diff
@@ -0,0 +1,220 @@
...
+            System.exit(0);
+        }
+        if (auth == null) {
+            System.out.println("auth not empty , please set --u=username:password");
```
Comment (chaoyli) [inline tools/stream_load/StreamLoadImportUtils.java:72]: auth is empty

File: tools/stream_load/StreamLoadImportUtils.java [hunk 3]
```diff
@@ -0,0 +1,220 @@
...
+            System.exit(0);
+        }
+        if (sourceFilePath == null) {
```
Comment (chaoyli) [inline tools/stream_load/StreamLoadImportUtils.java:75]: source file path is empty

File: tools/stream_load/README.md [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+
+Necessary:
+
+- --source-file: the absolute path of import file
+- --url: the `fe` url , it should contain protocol and so on, not `be` redirect url
+- --u: the starrocks database user and password ,not server
+
+Optional:
+
+- --is-debug: default false ,`--is-debug=true`
```
Comment (chaoyli) [inline tools/stream_load/README.md:24]: enable_debug

File: tools/stream_load/README.md [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+- --thread-number: parallel thread number , default min(server_core_number,32) ,`--thread-number=16`
+- --timeout: http protocol connect timeout and proccess timeout , default 60*1000ms, for example 5s `--timeout=5000`
+- --size: memory queue limit size , default 4096 , not reset unless you have a lot of memory
```
Comment (chaoyli) [inline tools/stream_load/README.md:27]: queue_size

File: tools/stream_load/README.md [hunk 6]
```diff
@@ -0,0 +1,39 @@
...
+Necessary:
+
+- --source-file: the absolute path of import file
+- --url: the `fe` url , it should contain protocol and so on, not `be` redirect url
+- --u: the starrocks database user and password ,not server
+
+Optional:
+
+- --is-debug: default false ,`--is-debug=true`
+- --thread-number: parallel thread number , default min(server_core_number,32) ,`--thread-number=16`
```
Comment (chaoyli) [inline tools/stream_load/README.md:25]: max_threads

**Review by chaoyli — APPROVED** [review]

**Review by gengjun-git — APPROVED** [review]

**Pull request merged**